jsonrpc/client.go

```diff
@@ -53,37 +53,69 @@ func (sc *Client) AccountNonce(context.Context, string) (*big.Int, error) {
 	panic("not implemented")
 }
 
+func (sc *Client) BlockNumber(ctx context.Context) (*big.Int, error) {
+	var blockNumber big.Int
+	if err := sc.c.CallContext(ctx, &blockNumber, "starknet_blockNumber"); err != nil {
+		return nil, err
+	}
+
+	return &blockNumber, nil
+}
+
 func (sc *Client) BlockByHash(ctx context.Context, hash string, scope string) (*types.Block, error) {
-	return sc.getBlock(ctx, "starknet_getBlockByHash", hash, scope)
+	var block types.Block
+	if err := sc.do(ctx, "starknet_getBlockByHash", &block, hash, scope); err != nil {
+		return nil, err
+	}
+
+	return &block, nil
 }
 
 func (sc *Client) BlockByNumber(ctx context.Context, number *big.Int, scope string) (*types.Block, error) {
-	return sc.getBlock(ctx, "starknet_getBlockByNumber", toBlockNumArg(number), scope)
+	var block types.Block
+	if err := sc.do(ctx, "starknet_getBlockByNumber", &block, toBlockNumArg(number), scope); err != nil {
+		return nil, err
+	}
+
+	return &block, nil
+}
+
+func (sc *Client) CodeByAddress(ctx context.Context, address string) (*types.Code, error) {
```

Comment:
Lets align with the go-ethereum interface as much as possible: https://pkg.go.dev/github.com/obscuren/go-ethereum/ethclient?utm_source=godoc#Client.CodeAt

```suggestion
func (sc *Client) CodeAt(ctx context.Context, address string) (*types.Code, error) {
```